tests/test_flat.py

```diff
@@ -0,0 +1,117 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+import os
+import numpy as np
+import unittest
+
+import lsst.afw.math as afwMath
+import lsst.daf.persistence as dafPersist
+import lsst.ip.isr as ipIsr
+import lsst.meas.algorithms as measAlg
+import lsst.utils.tests
+
+from lsst.pipe.tasks.repair import RepairTask
+
+
+# TODO: DM-26396
+#       Update these tests to validate calibration construction.
+class FlatTestCases(lsst.utils.tests.TestCase):
+
+    def setUpClass(self):
+        """Setup butler and generate an ISR processed exposure.
+
+        As no flat tests are described in DMTN-101, use similar tests
+        to those defined for darks.
+
+        Notes
+        -----
+        DMTN-101 10.X:
+
+        Process an independent flat frame through the ISR including
+        overscan correction, bias subtraction, dark subtraction, flat
+        correction
+
+        """
+        repoDir = os.path.join("..", "DATA")
+        calibDir = os.path.join("..", "DATA", "calibs")
+        butler = dafPersist.Butler(repoDir, calibRoot=calibDir)
+
+        self.config = ipIsr.IsrTaskConfig()
+        self.config.doSaturation = True
+        self.config.doSuspect = True
+        self.config.doSetBadRegions = True
+        self.config.doOverscan = True
+        self.config.doBias = True
+        self.config.doVariance = True
+        self.config.doDark = True
+        self.config.doFlat = True
+
+        self.config.doLinearize = False
+        self.config.doCrosstalk = False
+        self.config.doWidenSaturationTrails = False
+        self.config.doBrighterFatter = False
+        self.config.doDefect = False
+        self.config.doSaturationInterpolation = False
+        self.config.doStrayLight = False
+        self.config.doApplyGains = False
+        self.config.doFringe = False
+        self.config.doMeasureBackground = False
+        self.config.doVignette = False
+        self.config.doAttachTransmissionCurve = False
+        self.config.doUseOpticsTransmission = False
+        self.config.doUseFilterTransmission = False
+        self.config.doUseSensorTransmission = False
+        self.config.doUseAtmosphereTransmission = False
+
+        self.isrTask = ipIsr.IsrTask(config=self.config)
+        # This is not an independent frame.
+        self.dataRef = butler.dataRef('raw', dataId={'detector': 0, 'expId': 2020012800028})
+        results = self.isrTask.runDataRef(self.dataRef)
+        self.exposure = results.outputExposure
+
+    def test_independentFrameLevel(self):
+        """Test image mean and sigma are plausible.
+
+        Notes
+        -----
+        DMTN-101 10.X:
+        """
+        mean = afwMath.makeStatistics(self.exposure.getImage(), afwMath.MEAN).getValue()
+        sigma = afwMath.makeStatistics(self.exposure.getImage(), afwMath.STDEV).getValue()
+        expectMean = 8750
+        expectSigmaMax = 3000
```

Comment:
I guess these are hardcoded for the particular exposure (though sigma very seems high, no?). Will these need changing here by hand when we change data? Not arguing that that's a bad thing - I think we tend towards overengineering stuff tbh, and doing that here feels reasonable we'll change datasets quite rarely, and always by hand with human intervention, just checking that this is what's happening here.